File: SDKs/amazonaws.com/location/2020-11-19/python/src/sdk/models/shared/searchplaceindexforpositionsummary.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class SearchPlaceIndexForPositionSummary:
    r"""SearchPlaceIndexForPositionSummary
    A summary of the reverse geocoding request sent using <code>SearchPlaceIndexForPosition</code>.
    """
    
    data_source: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('DataSource') }})
    position: List[float] = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Position') }})
    max_results: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MaxResults') }})
    

File: SDKs/amazonaws.com/location/2020-11-19/typescript/src/sdk/models/shared/searchplaceindexforpositionsummary.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// SearchPlaceIndexForPositionSummary
/** 
 * A summary of the reverse geocoding request sent using <code>SearchPlaceIndexForPosition</code>.
**/
export class SearchPlaceIndexForPositionSummary extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=DataSource" })
  dataSource: string;

  @SpeakeasyMetadata({ data: "json, name=MaxResults" })
  maxResults?: number;

  @SpeakeasyMetadata({ data: "json, name=Position" })
  position: number[];
}
